pyfixest/estimation/decomposition.py

```diff
@@ -0,0 +1,269 @@
+
+    def fit(self, X: np.ndarray, Y: np.ndarray, store: bool = True):
+        "Fit Linear Mediation Model."
+        if store:
+            self.X = X
+            self.N = X.shape[0]
+            self.X1 = self.X[:, ~self.mask]
+            self.X1 = np.concatenate([np.ones((self.N, 1)), self.X1], axis=1)
+            self.names_X1 = ["Intercept", self.param]
+            self.param_in_X1_idx = self.names_X1.index(self.param)
+
+            self.X2 = self.X[:, self.mask]
+            self.Y = Y
+
+            self.direct_effect = np.linalg.lstsq(self.X1, self.Y, rcond=None)[
+                0
+            ].flatten()[self.param_in_X1_idx]
+            self.direct_effect = np.array([self.direct_effect])
+
+            self.gamma = np.linalg.lstsq(self.X1[:, 1:], self.X2)[0].flatten()
+            self.beta_full = np.linalg.lstsq(self.X, self.Y)[0].flatten()
+            self.beta2 = self.beta_full[self.mask].flatten()
+            self.delta = self.gamma * self.beta2.flatten()
+
+            if self.combine_covariates is not None:
+                for name, covariates in self.combine_covariates.items():
+                    variable_idx = [
+                        self.mediator_names.index(cov) for cov in covariates
+                    ]
+                    self.contribution_dict[name] = np.sum(self.delta[variable_idx])
+
+            self.contribution_dict["explained_effect"] = np.sum(
+                list(self.contribution_dict.values()), keepdims=True
+            ).flatten()
+            self.contribution_dict["unexplained_effect"] = (
+                self.direct_effect - self.contribution_dict["explained_effect"]
+            ).flatten()
+            self.contribution_dict["direct_effect"] = self.direct_effect
+            self.contribution_dict["full_effect"] = self.beta_full[
+                self.param_in_X1_idx
+            ].flatten()
+
+            # prepare bootstrap in first iteration
+            if self.unique_clusters is not None:
+                for g in self.unique_clusters:
+                    cluster_idx = np.where(self.cluster_df == g)[0]
+                    self.X_dict[g] = self.X[cluster_idx]
+                    self.Y_dict[g] = self.Y[cluster_idx]
+
+            return self.contribution_dict
+
+        else:
+            # Gelbach Method:
+
+            contribution_dict = {
+                key: 0.0 for key in self.combine_covariates_dict if key != "Intercept"
+            }
+
+            X1 = np.concatenate([np.ones((self.N, 1)), X[:, ~self.mask]], axis=1)
+            X2 = X[:, self.mask]
+
+            direct_effect = np.linalg.lstsq(X1, Y, rcond=None)[0].flatten()[
+                self.param_in_X1_idx
+            ]
+            direct_effect = np.array([direct_effect])
+
+            gamma = np.linalg.lstsq(X1[:, 1:], X2, rcond=None)[0].flatten()
+            beta_full = np.linalg.lstsq(X, Y, rcond=None)[0].flatten()
+            beta2 = beta_full[self.mask].flatten()
+            delta = gamma * beta2
```

Comment:
what's the difference between the mediation setup and gelbach's method setup? I think you run the same regressions?

Reply:
Yes I think both are identical (though I should take another look at the Ding chapter you mentioned earlier). I was thinking about adding yet another method, `mediate()`, which just runs `decompose()` under the hood? Might help to "un-confuse" users despite being the same thing? I.e. I would add 

```python 
def mediate(params): 
    self.decompose(params)
```
?

Reply:
Sure that seems fine (although strictly speaking decompose is a subset of mediate :) ). Just wanted to make sure i wasn't missing something subtle.